diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp
--- avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2016-11-20 16:53:38.579062697 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2016-11-20 19:53:35.913602582 +0100
@@ -18,8 +18,8 @@
 #include "ADM_ad_plugin.h"
 
 extern "C" {
-#include "ADM_liba52/a52.h"
-#include "ADM_liba52/mm_accel.h"
+#include "a52dec/a52.h"
+#include "a52dec/mm_accel.h"
 };
 
 #define AC3_HANDLE ((a52_state_t *)ac3_handle)
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt
--- avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2016-11-20 16:53:38.579062697 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2016-11-20 19:57:02.350079716 +0100
@@ -1,11 +1,9 @@
 INCLUDE(ad_plugin)
 
-ADD_SUBDIRECTORY(ADM_liba52)
-
 SET(ADM_ad_a52_SRCS ADM_ad_a52.cpp)
 
 ADD_AUDIO_DECODER( ADM_ad_a52  ${ADM_ad_a52_SRCS})
-TARGET_LINK_LIBRARIES(ADM_ad_a52 ADM_liba52)
+TARGET_LINK_LIBRARIES(ADM_ad_a52 ${LIBA52_LIBRARIES})
 
 INIT_AUDIO_PLUGIN(ADM_ad_a52)
 INSTALL_AUDIODECODER(ADM_ad_a52)
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp
--- avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp	2016-11-20 16:53:38.579062697 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp	2016-11-20 19:53:35.914602594 +0100
@@ -16,7 +16,7 @@
  ***************************************************************************/
 #include "ADM_default.h"
 #include "ADM_ad_plugin.h"
-#include "ADM_libMad/mad.h"
+#include "mad.h"
 
 #define Stream ((mad_stream *)_stream)
 #define Frame ((mad_frame *)_frame)
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt
--- avidemux_2.6.15.old/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt	2016-11-20 16:53:38.579062697 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt	2016-11-20 19:57:38.861513300 +0100
@@ -10,13 +10,11 @@
 	ADD_DEFINITIONS("-DFPM_ARM")
 ENDIF (ADM_CPU_X86_32)
 
-ADD_SUBDIRECTORY(ADM_libMad)
-
 ADD_DEFINITIONS("-DHAVE_ASSERT_H")
 SET(ADM_ad_Mad_SRCS ADM_ad_mad.cpp)
 
 ADD_AUDIO_DECODER( ADM_ad_Mad ${ADM_ad_Mad_SRCS})
-TARGET_LINK_LIBRARIES(ADM_ad_Mad ADM_libMad)
+TARGET_LINK_LIBRARIES(ADM_ad_Mad ${LIBMAD_LIBRARIES})
 
 INIT_AUDIO_PLUGIN(ADM_ad_Mad)
 INSTALL_AUDIODECODER(ADM_ad_Mad)
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt avidemux_2.6.15/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt
--- avidemux_2.6.15.old/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt	2016-11-20 16:53:38.568062568 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt	2016-11-21 14:43:16.431749948 +0100
@@ -6,11 +6,8 @@
 	muxerMp4v2Video.cpp
 	muxerMp4v2Config.cpp
 )
-SUBDIRS(libmp4v2)
-
 ADD_LIBRARY(ADM_mx_mp4v2 SHARED ${ADM_mp4v2_SRCS})
-TARGET_LINK_LIBRARIES(ADM_mx_mp4v2 ADM_libmp4v2)
+TARGET_LINK_LIBRARIES(ADM_mx_mp4v2 ${LIBMP4V2_LIBRARIES})
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/libmp4v2/include/)
 INIT_MUXER(ADM_mx_mp4v2)
 INSTALL_MUXER(ADM_mx_mp4v2)
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp avidemux_2.6.15/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp
--- avidemux_2.6.15.old/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp	2016-11-20 16:53:38.624063226 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp	2016-11-20 19:53:35.914602594 +0100
@@ -24,7 +24,7 @@
 #include "prefs.h"
 extern "C"
 {
-#include "ADM_libass/ass.h"
+#include "ass/ass.h"
 }
 
 /**
diff -Naur avidemux_2.6.15.old/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt avidemux_2.6.15/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt
--- avidemux_2.6.15.old/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt	2016-11-20 16:53:38.622063202 +0100
+++ avidemux_2.6.15/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt	2016-11-20 20:04:48.901620075 +0100
@@ -5,7 +5,6 @@
 checkFridibi("0.19")
 
 IF (USE_FREETYPE AND USE_FRIDIBI)
-	ADD_SUBDIRECTORY(ADM_libass)
 
 	INCLUDE(vf_plugin)
 	SET(ADM_vf_ssa_SRCS ADM_vidASS.cpp)
@@ -13,7 +12,7 @@
 	ADD_VIDEO_FILTER(ADM_vf_ssa ${ADM_vf_ssa_SRCS})
 
 	IF(DO_COMMON)
-		TARGET_LINK_LIBRARIES(ADM_vf_ssa ADM_libass ${FREETYPE2_LDFLAGS} ${FRIDIBI_LDFLAGS} ADM_coreSubtitle)
+                TARGET_LINK_LIBRARIES(ADM_vf_ssa ${LIBASS_LIBRARIES} ${FREETYPE2_LDFLAGS} ADM_coreSubtitle)
                 # We force the use of fontconfig
                 ADD_DEFINITIONS("-DUSE_FONTCONFIG")
                 TARGET_LINK_LIBRARIES(ADM_vf_ssa ${FONTCONFIG_LDFLAGS})
diff -Naur avidemux_2.6.15.old/avidemux_plugins/CMakeLists.txt avidemux_2.6.15/avidemux_plugins/CMakeLists.txt
--- avidemux_2.6.15.old/avidemux_plugins/CMakeLists.txt	2016-11-20 16:53:38.578062686 +0100
+++ avidemux_2.6.15/avidemux_plugins/CMakeLists.txt	2016-11-21 15:56:07.745137257 +0100
@@ -64,6 +64,24 @@
 LINK_DIRECTORIES("${AVIDEMUX_SEARCH_LIB_DIR}")
 INCLUDE_DIRECTORIES("${AVIDEMUX_SEARCH_INCLUDE_DIR}/avidemux/2.6")
 
+INCLUDE(FindPkgConfig)
+
+PKG_CHECK_MODULES(LIBASS REQUIRED libass)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBASS_INCLUDE_DIRS})
+
+PKG_CHECK_MODULES(LIBMAD REQUIRED mad)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBMAD_INCLUDE_DIRS})
+
+FIND_PATH(LIBA52_INCLUDE_DIR a52dec/a52.h)
+FIND_LIBRARY(LIBA52_LIBRARIES a52)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBA52_INCLUDE_DIR})
+
+PKG_CHECK_MODULES(LIBTWOLAME REQUIRED twolame)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBTWOLAME_INCLUDE_DIRS})
+
+FIND_PATH(LIBMP4V2_INCLUDE_DIR mp4v2/mp4v2.h)
+FIND_LIBRARY(LIBMP4V2_LIBRARIES mp4v2)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBMP4V2_INCLUDE_DIRS})
 
 IF (FRESH_BUILD)
 	MESSAGE("")
