diff -Naur avidemux_2.6.16.old/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt avidemux_2.6.16/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt
--- avidemux_2.6.16.old/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt	2017-01-04 14:05:47.560683107 +0100
+++ avidemux_2.6.16/avidemux_plugins/ADM_muxers/muxerMp4v2/CMakeLists.txt	2017-01-04 14:30:14.412052106 +0100
@@ -6,11 +6,8 @@
 	muxerMp4v2Video.cpp
 	muxerMp4v2Config.cpp
 )
-SUBDIRS(libmp4v2)
-
 ADD_LIBRARY(ADM_mx_mp4v2 SHARED ${ADM_mp4v2_SRCS})
-TARGET_LINK_LIBRARIES(ADM_mx_mp4v2 ADM_libmp4v2)
+TARGET_LINK_LIBRARIES(ADM_mx_mp4v2 ${LIBMP4V2_LIBRARIES})
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/libmp4v2/include/)
 INIT_MUXER(ADM_mx_mp4v2)
 INSTALL_MUXER(ADM_mx_mp4v2)
diff -Naur avidemux_2.6.16.old/avidemux_plugins/CMakeLists.txt avidemux_2.6.16/avidemux_plugins/CMakeLists.txt
--- avidemux_2.6.16.old/avidemux_plugins/CMakeLists.txt	2017-01-04 14:05:47.559683092 +0100
+++ avidemux_2.6.16/avidemux_plugins/CMakeLists.txt	2017-01-04 14:31:05.753833775 +0100
@@ -42,6 +42,14 @@
 endif(NOT EXISTS "${ADM_CMAKE_DIR}/commonCmakeApplication.cmake")
 MESSAGE(STATUS "Found cmake subfolder.good.")
 
+INCLUDE(FindPkgConfig)
+
+PKG_CHECK_MODULES(LIBTWOLAME REQUIRED twolame)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBTWOLAME_INCLUDE_DIRS})
+
+FIND_PATH(LIBMP4V2_INCLUDE_DIR mp4v2/mp4v2.h)
+FIND_LIBRARY(LIBMP4V2_LIBRARIES mp4v2)
+INCLUDE_DIRECTORIES(SYSTEM ${LIBMP4V2_INCLUDE_DIRS})
 
 ########################################
 # Where is the top dir ?
